src/runtime/pythonengine.cs

```diff
@@ -491,6 +491,452 @@ public enum RunFlagType
+            using (var pyKey = new PyString(name))
+            {
+                int r = Runtime.PyObject_DelItem(locals, pyKey.obj);
+                if (r < 0)
+                {
+                    throw new PythonException();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Exists Method
+        /// </summary>
+        /// <remarks>
+        /// Returns true if the variable appears in the local variable dict or the global variable dict.
+        /// </remarks>
+        public bool ContainsVariable(string name)
+        {
+            this.AcquireLock();
+            using (var pyKey = new PyString(name))
+            {
+                if (Runtime.PyMapping_HasKey(locals, pyKey.obj) != 0)
+                {
+                    return true;
+                }
+                return Runtime.PyMapping_HasKey(globals, pyKey.obj) != 0;
+            }
+        }
+
+        /// <summary>
+        /// Get Method
+        /// </summary>
+        /// <remarks>
+        /// Returns the value of the variable, local variable first.
+        /// If the variable is not exists, return null.
+        /// </remarks>
+        public PyObject GetVariable(string name)
+        {
+            this.AcquireLock();
+            using (var pyKey = new PyString(name))
+            {
+                IntPtr op;
+                if (Runtime.PyMapping_HasKey(locals, pyKey.obj) != 0)
+                {
+                    op = Runtime.PyObject_GetItem(locals, pyKey.obj);
+                }
+                else if (Runtime.PyMapping_HasKey(globals, pyKey.obj) != 0)
+                {
+                    op = Runtime.PyObject_GetItem(globals, pyKey.obj);
+                }
+                else
+                {
+                    return null; //name not exists
+                }
+                if (op == IntPtr.Zero)
+                {
+                    throw new PythonException();
+                }
+                return new PyObject(op);
+            }
+        }
+
+        public T GetVariable<T>(string name)
+        {
+            PyObject obj = this.GetVariable(name);
+            return (T)obj.AsManagedObject(typeof(T));
+        }
+
+        private static IntPtr GetInstHandle(object value)
+        {
+            if (value == null)
+            {
+                Runtime.XIncref(Runtime.PyNone);
+                return Runtime.PyNone;
+            }
+            else
+            {
+                var ptr = Converter.ToPython(value, value.GetType());
+                return ptr;
+            }
+        }
+
+        private void AcquireLock()
+        {
+            if(isDisposed)
+            {
+                throw new PySessionDisposedException();
+            }
+            this.state.AcquireLock();
+        }
+
+        public virtual void Dispose()
+        {
+            if(isDisposed)
+            {
+                return;
+            }
+            Runtime.XDecref(globals);
```

Comment:
`gc` issue is most likely here. You'll need to check that the engine is initialize and that it isn't finalizing before running these.